packages/vuelidate/src/core.js

```diff
@@ -476,22 +477,25 @@ export function setValidations ({
    */
 
   const $model = key ? computed({
-    get: () => unwrap(state[key]),
+    get: () => unwrap(unwrap(state)[key]),
     set: val => {
       $dirty.value = true
-      if (isRef(state[key])) {
-        state[key].value = val
+      const unwrappedState = unwrap(state)
+
+      if (isRef(unwrappedState[key])) {
+        unwrappedState[key].value = val
       } else {
-        state[key] = val
+        unwrappedState[key] = val
       }
     }
   }) : null
 
   if (config.$autoDirty) {
-    const watchTarget = isRef(state[key]) ? state[key] : computed(() => unwrap(state)[key])
-    watch(watchTarget, () => {
-      if (!$dirty.value) $touch()
-    })
+    watch(
+      () => unwrap(unwrap(state)[key]),
```

Comment:
This is already a computed, because we wrapped it earlier, correct?

Reply:
Yes, if we were using typescript `state` would be of type `T | Ref<T> | Record<keyof T, Ref<T[keyof T]>>` everywhere.